internal/cache/file/downloader/job_test.go

```diff
@@ -16,34 +16,87 @@ package downloader
 
 import (
 	"container/list"
+	"context"
 	"fmt"
+	"os"
+	"path"
 	"reflect"
 	"testing"
 
 	"github.com/googlecloudplatform/gcsfuse/internal/cache/data"
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/lru"
 	"github.com/googlecloudplatform/gcsfuse/internal/locker"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/fake"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/storageutil"
 	. "github.com/jacobsa/ogletest"
+	"github.com/jacobsa/timeutil"
 )
 
 ////////////////////////////////////////////////////////////////////////
 // Boilerplate
 ////////////////////////////////////////////////////////////////////////
 
+const TestBucketName = "test_bucket"
+const CacheMaxSize = 50
+const DefaultObjectName = "foo"
+
 func TestJob(t *testing.T) { RunTests(t) }
 
 type jobTest struct {
-	job *Job
+	job    *Job
+	bucket gcs.Bucket
+	cache  *lru.Cache
 }
 
 func init() { RegisterTestSuite(&jobTest{}) }
 
 func (jt *jobTest) SetUp(*TestInfo) {
 	locker.EnableInvariantsCheck()
-	var fileSpec data.FileSpec
-	jt.job = NewJob(nil, nil, nil, 200, fileSpec)
+
+	jt.bucket = fake.NewFakeBucket(timeutil.RealClock(), TestBucketName)
```

Comment:
Should we move

Reply:
We decided offline to keep it this way for now. (just use fake storage server bucket instead)